File: app/src/main/java/me/ykrank/s1next/data/api/model/UserSearchResult.java
package me.ykrank.s1next.data.api.model;

import com.google.common.base.Objects;

/**
 * Created by ykrank on 2017/04/01.
 */

public class UserSearchResult {

    private String uid;
    private String name;

    public String getUid() {
        return uid;
    }

    public void setUid(String uid) {
        this.uid = uid;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        UserSearchResult that = (UserSearchResult) o;
        return Objects.equal(uid, that.uid) &&
                Objects.equal(name, that.name);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(uid, name);
    }

    @Override
    public String toString() {
        return "UserSearchResult{" +
                "uid='" + uid + '\'' +
                ", name='" + name + '\'' +
                '}';
    }
}

File: app/src/main/java/me/ykrank/s1next/data/api/model/search/UserSearchResult.kt
package me.ykrank.s1next.data.api.model.search

import com.github.ykrank.androidtools.ui.adapter.StableIdModel

/**
 * Created by ykrank on 2017/04/01.
 */

class UserSearchResult : StableIdModel, SearchResult() {

    var uid: String? = null
    var name: String? = null

    override fun toString(): String {
        return "UserSearchResult{" +
                "uid='" + uid + '\''.toString() +
                ", name='" + name + '\''.toString() +
                '}'.toString()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as UserSearchResult

        if (uid != other.uid) return false
        if (name != other.name) return false

        return true
    }

    override fun hashCode(): Int {
        var result = uid?.hashCode() ?: 0
        result = 31 * result + (name?.hashCode() ?: 0)
        return result
    }
}
